Events Backend Cleanup and Adding Start/End Range test

adam-james-v opened the pull request:
This is a small PR to add some tests to card and collection timeline endpoints.

As well, some ;; todo items are cleaned up.

adam-james-v committed:
Adding tests in collections and card. Fixing some minor test things

adam-james-v committed:
Added Test for timeline event range selection via card API

adam-james-v committed:
Cleaned up a few TODO comments

adam-james-v committed:
Made the timestamp required when POST-ing a timeline-event

dpsutton commented on test/metabase/api/collection_test.clj:
```diff
@@ -1380,6 +1380,60 @@
+    (mt/with-temp* [Collection [coll-a {:name "Collection A"}]
+                    Collection [coll-b {:name "Collection B"}]
+                    Collection [coll-c {:name "Collection C"}]]
+    (mt/with-temp* [Timeline [tl-a {:name          "Timeline A"
```
You can just use a single `with-temp*` block, no need for the two nested ones.

adam-james-v replied:
Good catch, thanks. Fixed here and in cards test too

adam-james-v committed:
Remove unnecessary nesting of with-temp* forms

dpsutton commented on src/metabase/api/timeline.clj:
```diff
@@ -15,13 +15,16 @@
   "Events Query Parameters Schema"
   (s/enum "events"))
 
+(def Icons
```
I remember thinking about this location a bit. This seems like it specs the model not just the api. I think it should live in whichever model does not depend on the other model

adam-james-v committed:
Moved Icons schema into timeline-event model

This schema seems to be validating more than just the API, so we've moved it.

adam-james-v committed:
Added start/end to GET /api/timeline/:id

Also cleaned up some unnecessary nesting in a card API test
Added timeline range test in timeline API test ns too.

adam-james-v committed:
Parse timestamp when it exists in a timeline-event PUT

We saw the db error presented in the FE when updating timeline events. This should fix that problem.

adam-james-v committed:
Add a test to ensure we can change timestamps via timeline-event PUT

dpsutton commented on test/metabase/api/timeline_test.clj:
```diff
@@ -127,7 +178,7 @@
                                                 :archived    true}]]
           (testing "a timeline with no events returns an empty list"
-            (is (= #{} (event-names (include-events-request empty-tl false)))))
+            (is (= '() (:events (include-events-request empty-tl false)))))
```
no need to change now but you don't need to quote the empty list in Clojure. And equality will check across collection type so you could use a more common `[]`. or just assert that it is `empty?`

adam-james-v commented:
This was some cleanup for the Epic #20289
